rocketpy/Components.py

```diff
@@ -0,0 +1,125 @@
+__copyright__ = "Copyright 20XX, RocketPy Team"
+__license__ = "MIT"
+
+from collections import namedtuple
+
+
+class Components:
+    """A class to hold components of the Rocket class. A component is
+    an object that added to the rocket given a position relative to the
+    rocket's coordinate system origin. Components can be added to the
+    rocket using the 'add' methods. This class is currently used only to
+    hold aerodynamic surfaces.
+
+    Attributes
+    ----------
+    _components: list
+        A list of named tuples containing the component and its position.
+    component_tuple: namedtuple
+        A named tuple containing the component and its position.
+    """
+
+    def __init__(self):
+        """Initialize an empty components list instance."""
+        self.component_tuple = namedtuple("component_tuple", "component position")
+        self._components = []
+
+    def __repr__(self):
+        """Return a string representation of the Components instance."""
+        return repr(self._components)
+
+    def __len__(self):
+        """Return the number of components in the collection."""
+        return len(self._components)
+
+    def __getitem__(self, index):
+        """Return the number of components in the collection."""
+        return self._components[index]
+
+    def __iter__(self):
+        """Return an iterator over the components."""
+        return iter(self._components)
+
+    def add(self, component, position):
+        """Add a component to the collection.
+
+        Parameters
+        ----------
+            component: AeroSurface, Motor
+                The component to be added to the rocket.
+            position: int, float
+                The position of the component relative to the rocket's
+                coordinate system origin.
+
+        Returns
+        -------
+        None
+        """
+        self._components.append(self.component_tuple(component, position))
+
+    def get_by_type(self, component_type):
+        """Get the components of a specified given type.
+
+        Parameters
+        ----------
+        component_type: type
+            The type of component to be returned.
+
+        Returns:
+        --------
+        list
+            A list of components matching the specified type.
+
+        """
+        component_type_list = [
+            c.component
+            for c in self._components
+            if isinstance(c.component, component_type)
+        ]
+        return component_type_list
+
+    def remove(self, component):
+        """Remove a component from the list of components.
+
+        Parameters
+        ----------
+        component: AeroSurface, Motor
+            The component to be removed from the rocket.
+
+        Returns
+        --------
+        None
+        """
+        for index, comp in enumerate(self._components):
+            if comp.component == component:
+                self._components.pop(index)
+                break
```

Comment:
What would happen if the same component is added twice in the rocket's component list?

For instance, I can create a single fin and add it 3 times to the rocket. And if I want I can even set them at the very same position.

Shall we be worried about that?

Reply:
Suggestion in case you're tired and wants to move forward: just remove the break and let it search in the entire list XD

Reply:
I actually thought of that when writing the code. What I thought was that if the method is implemented to remove all the same components even if it is in the list more than once, the user would not be able to easily remove only one of those instances.

The way it is currently done, the user could just call `remove` as many times as needed and it is in their control.

By the way, this is how a standard Python list deals with `.remove`

Reply:
OK then, it makes a lot of sense to me now.
But I still think that we could mention, in the documentation, that it will remove only the first appearance of the object.

Reply:
Done in https://github.com/RocketPy-Team/RocketPy/pull/355/commits/e53404f81bf89862181c1b8732331d01db6b10f0